File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
           int max_depth = 0;
        int current_depth = 0;
        
        for (char ch : s) {
            if (ch == '(') {
                current_depth++;
                max_depth = std::max(max_depth, current_depth);
            } else if (ch == ')') {
                current_depth--;
            }
        }
        
        return max_depth;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.ts
function maxDepth(s: string): number {
     let maxDepth = 0;
        let currentDepth = 0;

        for (const char of s) {
            if (char === '(') {
                currentDepth++;
                maxDepth = Math.max(maxDepth, currentDepth);
            } else if (char === ')') {
                currentDepth--;
            }
        }

        return maxDepth;
};